### PR TITLE
fix #2470 Make refguide BOM snippets use version from gradle.properties

### LINKED ISSUE #2470: Reference guide still mentions very old Bismuth-RELEASE BOM
In the reference guide, the sections guiding on how to add the BOM using Maven and Gradle still reference `Bismuth-RELEASE`.

The current latest BOM is `2020.0.0`, with `Dysprosium-SR13` a close second (and the one most people should have in prod right now).

The README of the project is updated on each release, but indeed the reference guide mention of the release train has been overlooked :(

**This issue targets the `3.3.x` branch, and as such the latest is `Dysprosium-SR13`**, to be updated at the next release to SR-14.

### PR DESCRIPTION
**Pull request by simonbasle**
Additional tasks once approved:
 - [ ] merge with squash, editing the original commit message to reflect new properties-based approach (and `Reviewed-in: #2474`)
 - [ ] ⚠️ merge the commit from 3.3.x into `master` with an edit:
   - [ ] `git merge --no-commit 3.3.x`
   - [ ] change the `bomVersion` attribute in `gradle.properties` to `2020.0.0`
   - [ ] `git commit`
   - [ ] push to master

### CODE REVIEW

**Comment by simonbasle** [comment]
This is what it looks like for a snapshot (see the warning note and the snippet):

image

File: gradle/releaser.gradle [hunk 1]
```diff
@@ -55,25 +55,40 @@ task copyReadme(type: Copy, group: "releaser helpers", description: "copies the
...
 	into rootProject.rootDir
-	filter { line -> line
-		.replace(oldVersion, currentVersion)
-		.replace(oldSnapshot, nextVersion)
+	filter { line -> (bomVersion == null) ?
+		line.replace(oldVersion, currentVersion)
+			.replace(oldSnapshot, nextVersion) :
+		line.replace(oldVersion, currentVersion)
+			.replace(oldSnapshot, nextVersion)
+			.replaceAll("\\:reactorReleaseTrain\\: .*", ":reactorReleaseTrain: " + bomVersion)
```
Comment (ericbottard) [inline gradle/releaser.gradle:91]: I suggest passing the attribute directly via the asciidoc plugin, ie https://github.com/reactor/reactor-core/blob/4bdfbd3bef81b1a2992057ceaf0892d4d22d823f/gradle/asciidoc.gradle#L44

Comment (simonbasle) [reply gradle/releaser.gradle:91]: let's put it in the `gradle.properties` then, and a manual step on top of the README bump task

Comment (simonbasle) [reply gradle/releaser.gradle]: updated

**Review by aneveu — APPROVED** [review]

File: docs/asciidoc/gettingStarted.adoc [hunk 2]
```diff
@@ -83,20 +83,25 @@ However, if you want to force the use of a specific artifact's version, you can
 it when adding your dependency, as you usually would. You can also forgo the BOM entirely
 and specify dependencies by their artifact versions.
 
+WARNING: As of this version (reactor-core {project-version}), the latest stable BOM in the associated
+release train line is `{reactorReleaseTrain}`, which is what is used in snippets below.
+There might be newer versions since then (including snapshots, milestones and new release train lines),
+see https://github.com/reactor/reactor/releases/.
```
Comment (bsideup) [inline docs/asciidoc/gettingStarted.adoc:89]: wouldn't it be better to link to https://projectreactor.io/docs that contains both the release train + core's version?

Comment (simonbasle) [reply docs/asciidoc/gettingStarted.adoc]: indeed, fixed

**Review by bsideup — APPROVED** [review]

**Review by aneveu — APPROVED** [review]

**Pull request merged**